File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String ret = "";
        boolean mode = false;
        for (int i = 0; i < code.length(); i++) {
            if (!mode) {
                if (code.charAt(i) != '1') {
                    if (i % 2 == 0) {
                        ret += String.valueOf(code.charAt(i));
                    }
                } else {
                    mode = (!mode);
                }
            } else {
                if (code.charAt(i) != '1') {
                    if (i % 2 != 0) {
                        ret += String.valueOf(code.charAt(i));
                    }
                } else {
                    mode = (!mode);
                }
            }
        }
        
        if (ret.isEmpty()) {
            return "EMPTY";
        } else {
            return ret;
        }
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    mode = 0
    ret = ''
    for i in range(len(code)):
        if mode == 0: #mode가 0일 때
            if code[i] == "1":
                mode = 1
            else:
                if i % 2 == 0:
                    ret += code[i]
                    i += 1
        else: #mode가 1일 때
            if code[i] == "1":
                mode = 0
            else:
                if i % 2 == 1:
                    ret += code[i]
                    i += 1
            
    if ret == "":
        return "EMPTY"
    else:
        return ret
